tools/update-eslint-recommended.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @fileoverview Script to update the eslint:recommended configuration.
+ * @author aladdin-add<weiran.zsd@outlook.com>
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const fs = require("node:fs");
+const builtInRules = require("../lib/rules");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const recommendedRule = {};
+
+for (const [ruleId, rule] of builtInRules) {
+	if (!rule.meta.deprecated) {
+		if (rule.meta.docs.recommended === true) {
+			recommendedRule[ruleId] = "error";
+		}
+	}
+}
+
+const recommendedRulesText = JSON.stringify(
+	recommendedRule,
+	null,
+	4,
+).replaceAll("\n", "\n    ");
+
+//-----------------------------------------------------------------------------
+// Main
+//-----------------------------------------------------------------------------
+
+const code = `/*
+ * WARNING: This file is autogenerated using the tools/update-eslint-recommended.js
+ * script. Do not edit manually.
+ */
+"use strict";
+
+/*
+ * IMPORTANT!
+ *
+ * We cannot add a "name" property to this object because it's still used in eslintrc
+ * which doesn't support the "name" property. If we add a "name" property, it will
+ * cause an error.
+ */
+
+module.exports = Object.freeze({
+	rules: Object.freeze(${recommendedRulesText}),
```

Comment:
```suggestion
    rules: Object.freeze(${recommendedRulesText}),
```

Since the rest of the generated file uses spaces, it would be consistent to have 4 spaces instead of a tab here.